index.js

```diff
@@ -52,7 +56,7 @@ module.exports = {
 
         this.log('preparing to upload revision to S3 bucket `' + bucket + '`', { verbose: true });
 
-        var s3 = new S3({ plugin: this });
+        var s3 = this.readConfig('s3DeployClient');
```

Comment:
this is for cleanup and to enable stubbing on unit tests for `index.js` 

we don't have an end-to-end test but this **should** not break anything /cc @LevelbossMike @lukemelia